### PR TITLE
Add custom icon functionality

### PR DESCRIPTION
**Pull request by NomisCZ**
Add icon setting to tags.
https://discuss.flarum.org/d/7078-change-tag-left-sidebar-icon-with-custom-font-awesome-icon/10
https://discuss.flarum.org/d/11467-add-fontawesome-icons-to-tags/9

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@ composer.phar
 Thumbs.db
 node_modules
 js/dist/*
+.idea
```
Comment (franzliedke) [inline .gitignore]: Please add this line (personal IDE preferences, basically) to your local [global gitignore file](https://help.github.com/en/articles/ignoring-files#create-a-global-gitignore) or to your repo's [excludes file](https://help.github.com/en/articles/ignoring-files#explicit-repository-excludes).

File: js/src/admin/components/EditTagModal.js [hunk 2]
```diff
@@ -19,6 +19,7 @@ export default class EditTagModal extends Modal {
     this.slug = m.prop(this.tag.slug() || '');
     this.description = m.prop(this.tag.description() || '');
```
Comment (franzliedke) [inline js/src/admin/components/EditTagModal.js]: Please don't commit changes to the dist JS files. These will be [changed by our bot](https://github.com/flarum/tags/commit/1d0ee59531749a869d3498f95365d2886238a147) once your changes are merged.

File: js/src/common/helpers/tagLabel.js [hunk 3]
```diff
@@ -1,10 +1,12 @@
 import extract from 'flarum/utils/extract';
+import tagIcon from "./tagIcon";
```
Comment (franzliedke) [inline js/src/common/helpers/tagLabel.js]: Please try to stay consistent with the code style in each file (in this case, the quotes used when there is no interpolation).

File: js/src/common/helpers/tagLabel.js [hunk 4]
```diff
@@ -21,11 +23,10 @@ export default function tagLabel(tag, attrs = {}) {
   } else {
     attrs.className += ' untagged';
   }
-
   return (
     m((link ? 'a' : 'span'), attrs,
       <span className="TagLabel-text">
-        {tag ? tag.name() : app.translator.trans('flarum-tags.lib.deleted_tag_text')}
+          {tag.icon() && tagIcon(tag, {}, { disableColors: true }) } {tagText}
```
Comment (franzliedke) [inline js/src/common/helpers/tagLabel.js]: Don't change indentation, please. ;)

I won't comment on other instances for now.

**Comment by franzliedke** [comment]
That said, is this a feature we want? @tobscure @luceos @jordanjay29

**Comment by jordanjay29** [comment]
It's been requested enough times and seems in line with Groups, I can see it being a nice addition. It might work better as an extension, but there's no norm yet for extensions of extensions (other than languages). So this works as an addition to Tags IMHO.

**Comment by luceos** [comment]
Yeah I think we need this. Is the icon optional and is the tag modal clear enough about that?

**Comment by tobscure** [comment]
I think this is reasonable to add

**Comment by franzliedke** [comment]
@NomisCZ Please review the complete changes in [the diff tab](https://github.com/flarum/tags/pull/60/files). There should not be any changes to .gitignore, and the JS dist files at all. Optimally, you should [amend the pushed commits](https://www.burntfen.com/2015-10-30/how-to-amend-a-commit-on-a-github-pull-request) so that they do not contain any changes to those files (instead of having multiple commits that first add and then revert the changes).

Thanks for your work on this! :raised_hands:

**Comment by NomisCZ** [comment]
@franzliedke I reverted all my previous commits and I added only my "actual" changes.

**Pull request merged**